lib/internal/child_process/serialization.js

```diff
@@ -131,16 +130,12 @@ const advanced = {
 const json = {
   initMessageChannel(channel) {
     channel[kJSONBuffer] = '';
-    channel[kStringDecoder] = undefined;
   },
 
   *parseChannelMessages(channel, readData) {
     if (readData.length === 0) return;
 
-    if (channel[kStringDecoder] === undefined)
-      channel[kStringDecoder] = new StringDecoder('utf8');
-    const chunks =
-      StringPrototypeSplit(channel[kStringDecoder].write(readData), '\n');
+    const chunks = StringPrototypeSplit(decodeUTF8(readData, false), '\n');
```

Comment:
The purpose of StringDecoder is streaming UTF-8 decoding; you cannot replace a streaming decoder with a non-streaming one without breaking its functionality, because the latter doesn't handle characters that aresplit across multiple bytes well.